server/service/src/apis/api_on_central/mod.rs

```diff
@@ -0,0 +1,122 @@
+use log::info;
+use repository::{NameStoreJoinRepository, NameStoreJoinRow, RepositoryError};
+use reqwest::{ClientBuilder, StatusCode};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use url::Url;
+use util::format_error;
+
+use crate::{
+    service_provider::{ServiceContext, ServiceProvider},
+    sync::CentralServerConfig,
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NameStoreJoinParams {
+    pub id: String,
+    pub name_id: String,
+    pub store_id: String,
+}
+
+#[derive(Debug)]
+pub struct SiteAuth {
+    pub username: String,
+    pub password_sha256: String,
+}
+
+#[derive(Deserialize, Debug, Error, Serialize)]
+pub enum CentralApiError {
+    #[error("Not a central server")]
+    NotACentralServer,
+    #[error("Not authorized")]
+    NotAuthorized,
+    #[error("Connection error: {0}")]
+    ConnectionError(String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+}
+
+impl From<RepositoryError> for CentralApiError {
+    fn from(from: RepositoryError) -> Self {
+        CentralApiError::InternalError(format_error(&from))
+    }
+}
+
+/// Creates/updates a name_store_join for a patient
+pub async fn patient_name_store_join(
+    service_provider: &ServiceProvider,
+    NameStoreJoinParams {
+        id,
+        name_id,
+        store_id,
+    }: NameStoreJoinParams,
```

Comment:
Remote site has already added/updated the name_store_join record on OG, and has sent us the result. We just save this